index.js

```diff
@@ -6,31 +6,26 @@ module.exports = (embark) => {
   if (embark.config.embarkConfig.versions.solc) {
     const versionPromise = new Promise(function(resolve, reject) {
       // Check solc version
+      
+    });
+
+    embark.registerCompiler('.sol', (contractFiles, options, cb) => {
+      if (!contractFiles || !contractFiles.length) {
+        return cb();
+      }
       Compiler.getSolcVersion(embark.logger, (err, version) => {
```

Comment:
This will make so that we check the version for each new compilation (of all the files). Is this ok? Can the wanted version change?